mmdet/models/roi_heads/sparse_roi_head.py

```diff
@@ -0,0 +1,313 @@
+                iterative process.
+            x (List[Tensor]): List of FPN features
+            rois (Tensor): Rois in total batch. With shape (num_proposal, 5).
+                the last dimension 5 represents (img_index, x1, y1, x2, y2).
+            object_feats (Tensor): The object feature extracted from
+                the previous stage.
+            img_metas (dict): meta information of images.
+
+        Returns:
+            dict[str, Tensor]: a dictionary of bbox head outputs,
+                Containing the following results:
+
+                    - cls_score (Tensor): The score of each class, has
+                      shape (batch_size, num_proposals, num_classes)
+                      when use focal loss or
+                      (batch_size, num_proposals, num_classes+1)
+                      otherwise.
+                    - decode_bbox_pred (Tensor): The regression results
+                      with shape (batch_size, num_proposal, 4).
+                      The last dimension 4 represents
+                      [tl_x, tl_y, br_x, br_y].
+                    - object_feats (Tensor): The object feature extracted
+                      from current stage
+                    - detach_cls_score_list (list[Tensor]): The detached
+                      classification results, length is batch_size, and
+                      each tensor has shape (num_proposal, num_classes).
+                    - detach_proposal_list (list[tensor]): The detached
+                      regression results, length is batch_size, and each
+                      tensor has shape (num_proposal, 4). The last
+                      dimension 4 represents [tl_x, tl_y, br_x, br_y].
+        """
+        num_imgs = len(img_metas)
+        bbox_roi_extractor = self.bbox_roi_extractor[stage]
+        bbox_head = self.bbox_head[stage]
+        bbox_feats = bbox_roi_extractor(x[:bbox_roi_extractor.num_inputs],
+                                        rois)
+        cls_score, bbox_pred, object_feats = bbox_head(bbox_feats,
+                                                       object_feats)
+        proposal_list = self.bbox_head[stage].refine_bboxes(
+            rois,
+            rois.new_zeros(len(rois)),  # dummy arg
+            bbox_pred.view(-1, bbox_pred.size(-1)),
+            [rois.new_zeros(object_feats.size(1)) for _ in range(num_imgs)],
+            img_metas)
+        bbox_results = dict(
+            cls_score=cls_score,
+            decode_bbox_pred=torch.cat(proposal_list),
+            object_feats=object_feats,
+            # detach then use it in label assign
+            detach_cls_score_list=[
+                cls_score[i].detach() for i in range(num_imgs)
+            ],
+            detach_proposal_list=[item.detach() for item in proposal_list])
+
+        return bbox_results
+
+    def forward_train(self,
+                      x,
+                      proposal_boxes,
+                      proposal_features,
```

Comment:
In forward_train, it is `proposal_features`, in simple_test, it is `proposal_feats`. Please unify the names of similar variables.